src/components/encryption.ts

```diff
@@ -357,18 +327,21 @@ export class EncryptedEventBroker {
     /**
      * Stop syncing clients used for encryption
      */
-    public close() {
+    public close(): void {
         for (const client of this.syncingClients.values()) {
-            client.stopClient();
-        }
-        if (this.presenceCleanupInterval) {
-            clearInterval(this.presenceCleanupInterval);
+            try {
+                client.matrixClient.stop();
+            }
+            catch (ex) {
+                // Non-fatal
+                log.warn(`MatrixClient failed to stop`, ex);
+            }
         }
     }
 
-    public static supportsLoginFlow(loginFlows: {flows: {type: string}[]}) {
+    public static supportsLoginFlow(loginFlows: {flows: {type: string}[]}): boolean {
         return loginFlows.flows.find(
             flow => flow.type === APPSERVICE_LOGIN_TYPE
-        );
+        ) !== undefined;
```

Comment:
I'd go for a `!!` prefix but this is fine.

Reply:
I'd go for `.findIndex() !== -1` :laughing: 
So many options.

Reply:
I've left it alone because I want to be contrarian :D